src/main/java/ru/practicum/shareit/user/service/UserServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package ru.practicum.shareit.user.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ru.practicum.shareit.exeption.ObjectNotFoundException;
+import ru.practicum.shareit.exeption.ValidationException;
+import ru.practicum.shareit.user.dto.UserDto;
+import ru.practicum.shareit.user.model.User;
+import ru.practicum.shareit.user.storage.UserStorage;
+
+import java.util.List;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class UserServiceImpl implements UserService {
+
+    private final UserStorage repository;
+
+    @Override
+    public List<User> findAll() {
+        log.info("Пользователи отправлены");
+        return repository.findAll();
+    }
+
+    @Override
+    public User getById(long id) {
+        log.info("Пользователь с id{} отправлен", id);
+        return repository.getById(id).orElseThrow(() -> {
+            log.warn("User with id {} not found", id);
+            throw new ObjectNotFoundException("User not found");
+        });
+    }
+
+    @Override
+    public UserDto create(UserDto userDto) {
+        validator(userDto.getEmail());
+        log.info("Пользователь создан");
+        return repository.create(userDto);
+    }
+
+    @Override
+    public User update(long id, User user) {
+        if (user.getEmail() != null) validator(user.getEmail());
+        return repository.update(id, user);
+    }
+
+    @Override
+    public void delete(long id) {
+        log.info("Пользователь с id {} удалён", id);
+        repository.delete(id);
+    }
+
+    private void validator(String email) {
+        List<User> users = repository.findAll();
+        boolean flag = users.stream().anyMatch(repoUser -> repoUser.getEmail().equals(email));
```

Comment:
Стрим лучше разбивать на несколько строк, чтобы код было проще читать:
```suggestion
        boolean flag = users.stream()
            .anyMatch(repoUser -> repoUser.getEmail().equals(email));
```

А еще его можно вынести в отдельный приватный метод, например:
`private boolean containsEmail(List<User> users, String email) `
И применять внутри условия:
`if (containsEmail(users, email))`

Это тоже повысит читаемость кода